src/views/AddItem.jsx

```diff
@@ -1,3 +1,113 @@
-export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+import { useState } from 'react';
+import { addItem } from '../api/firebase';
+
+export function AddItem({ listToken }) {
+	// Initialize formData state to store user input
+	const [formData, setFormData] = useState({
+		itemName: '',
+		daysUntilNextPurchase: 7,
+	});
+	// Initialize submission message to display after user submission
+	const [submissionStatus, setSubmissionStatus] = useState('');
+
+	// This handleChange function updates the formData object as we receive user inputs
+	// It converts radio button value from string to integer type for calculation of "nextPurchasedDate"
+	function handleChange(event) {
+		// destructure the event target
+		const { name, value, type } = event.target;
+		setFormData((prevFormData) => ({
+			...prevFormData,
+			[name]: type === 'radio' ? parseInt(value) : value,
+		}));
+	}
+
+	// This async handleSubmit function sends the data to firebase and displays a user message
+	async function handleSubmit(event) {
+		event.preventDefault();
+		// check if user has entered an empty string or whitespace
+		if (formData.itemName.trim() === '') {
+			setSubmissionStatus('Please enter an item name.');
+			return;
+		}
+
+		try {
+			await addItem(listToken, formData); // add formData to firebase
+			setSubmissionStatus(
+				`${formData.itemName} was successfully added to the database`,
+			);
+			setFormData({
+				itemName: '',
+				daysUntilNextPurchase: 7,
+			}); // clear the itemName field after submission
+		} catch (error) {
```

Comment:
I like that you used a `try catch` block here. Were you able to trigger the catch statement?